### PR TITLE
Jetpack Plugin: Update Tracking ajax callback

### PR DESCRIPTION
**Pull request by karthikax**
Update universal ajax callback for all tracking events to be defined in the tracking library rather than per consumer basis to avoid conflict between tracking library consumers.

Thank you so much @kbrown9 for the implementation.

#### Changes proposed in this Pull Request:

Tracking: set up the ajax hook in the Tracking class.
* Set up the ajax hook in the Tracking class. Use an action to ensure that the hook is set up only once.
* Since the hook is set up once, the event type prefix will be the product name of the object that set up the hook. This isn't the expected behavior. Fix that by adding a new parameter to the record_user_event() method, 'use_product_prefix'.
  This parameter is used to indicate whether the product name should be used as the prefix. If false, 'jetpack' is used.
* In the 'ajax_tracks' method, pass 'false' as the value of 'use_product_prefix' in the call to record_user_event(). For
  events that are tracked via Ajax, the prefix will always be 'jetpack_', as indicated in the README document.

Remove Jetpack specific Tracks ajax hook.
* Since ajax hook is defined in the Tracking class now, defining wp_ajax_jetpack_tracks in a consumer, creates duplication and hence removed.

#### Jetpack product discussion

p9dueE-3w6-p2

#### Does this pull request change what data or activity we track or use?

This changes the method with which we track. But does not change what data we track.

#### Testing instructions:

* Go to a site with Jetpack installed and activated.
* Navigate to wp-admin/admin.php?page=jetpack_modules.
* Enable or disable a module.
* Test for the corresponding event in the Tracks dashboard in ~5 to 7 minutes.

### CODE REVIEW

File: projects/packages/tracking/src/class-tracking.php [hunk 1]
```diff
@@ -47,6 +47,12 @@ public function __construct( $product_name = 'jetpack', $connection = null ) {
 			// TODO We should always pass a Connection.
 			$this->connection = new Connection\Manager();
 		}
+
+		if ( ! did_action( 'jetpack_set_tracks_ajax_hook' ) ) {
+			add_action( 'wp_ajax_jetpack_tracks', array( $this, 'ajax_tracks' ) );
+
+			do_action( 'jetpack_set_tracks_ajax_hook' );
```
Comment (kbrown9) [inline projects/packages/tracking/src/class-tracking.php:54]: This new action needs a [docblock](https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php/#4-hooks-actions-and-filters). I can add that if you'd like.

I don't think I should provide the approval for this PR, so we'll need to get someone else from Crew to approve it.

Comment (kbrown9) [reply projects/packages/tracking/src/class-tracking.php:61]: I had to update a `composer.lock` file for this PR, so I took the opportunity to add the docblock in commit 2aeeb922

**Review by leogermani — APPROVED** [review]
Looks and tests well for me.

(I tested using the upgrade nudges buttons in the Stats page, as they use the ajax action we changed in this PR and made sure the action was properly added and the events were being tracked)

**Comment by samiff** [comment]
cherry-picked to jetpack/branch-10.2 in c463900157f6c122dc706331ca46f8a5c41393e0

v1.13.11 of packages/tracking will release once https://github.com/Automattic/jetpack/pull/21254 lands & processes.

**Comment by samiff** [comment]
r232823-wpcom

**Pull request merged**